src/test/java/FreeTimeServletTest.java

```diff
@@ -0,0 +1,211 @@
+  private static final Date EVENT_THREE_END = new Date(2020 - OFFSET_YEAR, 4, 19, 10, 30);
+  private static final EventDateTime START_ONE =
+      new EventDateTime().setDateTime(new DateTime(EVENT_ONE_START));
+  private static final EventDateTime END_ONE =
+      new EventDateTime().setDateTime(new DateTime(EVENT_ONE_END));
+  private static final EventDateTime START_TWO =
+      new EventDateTime().setDateTime(new DateTime(EVENT_TWO_START));
+  private static final EventDateTime END_TWO =
+      new EventDateTime().setDateTime(new DateTime(EVENT_TWO_END));
+  private static final EventDateTime START_THREE =
+      new EventDateTime().setDateTime(new DateTime(EVENT_THREE_START));
+  private static final EventDateTime END_THREE =
+      new EventDateTime().setDateTime(new DateTime(EVENT_THREE_END));
+  private static final List<Event> NO_EVENT = ImmutableList.of();
+  private static final List<Event> EVENT_ONE =
+      ImmutableList.of(
+          new Event().setSummary(EVENT_SUMMARY_ONE).setStart(START_ONE).setEnd(END_ONE));
+  private static final List<Event> EVENT_TWO =
+      ImmutableList.of(
+          new Event().setSummary(EVENT_SUMMARY_TO_READ).setStart(START_TWO).setEnd(END_TWO));
+  private static final List<Event> EVENT_THREE =
+      ImmutableList.of(
+          new Event().setSummary(EVENT_SUMMARY_THREE).setStart(START_THREE).setEnd(END_THREE));
+
+  @Override
+  @Before
+  public void setUp() throws Exception {
+    super.setUp();
+    calendarClientFactory = Mockito.mock(CalendarClientFactory.class);
+    calendarClient = Mockito.mock(CalendarClient.class);
+    GmailClientFactory gmailClientFactory = Mockito.mock(GmailClientFactory.class);
+    gmailClient = Mockito.mock(GmailClient.class);
+    servlet =
+        new FreeTimeServlet(authenticationVerifier, calendarClientFactory, gmailClientFactory);
+    Mockito.when(gmailClientFactory.getGmailClient(Mockito.any())).thenReturn(gmailClient);
+    Mockito.when(calendarClientFactory.getCalendarClient(Mockito.any())).thenReturn(calendarClient);
+    // Writer used in get/post requests to capture HTTP response values
+    stringWriter = new StringWriter();
+
+    request = Mockito.mock(HttpServletRequest.class);
+    response =
+        Mockito.mock(
+            HttpServletResponse.class,
+            AdditionalAnswers.delegatesTo(new HttpServletResponseFake(stringWriter)));
+    Mockito.when(request.getCookies()).thenReturn(validCookies);
+  }
+
+  @Test
+  public void noPotentialEvent() throws Exception {
+    // Test case where there are no unread emails
+    // We should get no proposed potential event times
+    Mockito.when(calendarClient.getCalendarList()).thenReturn(ONE_CALENDAR);
+    Mockito.when(calendarClient.getUpcomingEvents(PRIMARY, CURRENT_TIME, END_TIME))
+        .thenReturn(NO_EVENT);
+    Mockito.when(calendarClient.getCurrentTime()).thenReturn(CURRENT_TIME);
+    Mockito.when(gmailClient.getWordCount(7)).thenReturn(0);
```

Comment:
What is the meaning of this `7`? I would consider changing this to a constant as this seems non-obvious

Reply:
`Resolved`